freqtrade/data/entryexitanalysis.py

```diff
@@ -0,0 +1,259 @@
+                       'profit_abs_median', 'profit_abs_mean', 'median_profit_pct',
+                       'mean_profit_pct', 'total_profit_pct']
+        sortcols = ['profit_abs_sum', 'enter_reason']
+
+        new['median_profit_pct'] = new['median_profit_pct'] * 100
+        new['mean_profit_pct'] = new['mean_profit_pct'] * 100
+        new['total_profit_pct'] = new['total_profit_pct'] * 100
+
+        _print_table(new, sortcols)
+    if "4" in glist:
+        new = bigdf.groupby(['pair', 'enter_reason', 'exit_reason']) \
+                   .agg({'profit_abs': ['count', 'sum', 'median', 'mean'],
+                         'profit_ratio': ['sum', 'median', 'mean']}
+                        ).reset_index()
+        new.columns = ['pair', 'enter_reason', 'exit_reason', 'num_buys', 'profit_abs_sum',
+                       'profit_abs_median', 'profit_abs_mean', 'median_profit_pct',
+                       'mean_profit_pct', 'total_profit_pct']
+        sortcols = ['profit_abs_sum', 'enter_reason']
+
+        new['median_profit_pct'] = new['median_profit_pct'] * 100
+        new['mean_profit_pct'] = new['mean_profit_pct'] * 100
+        new['total_profit_pct'] = new['total_profit_pct'] * 100
+
+        _print_table(new, sortcols)
+
+
+def _print_results(analysed_trades, stratname, group,
+                   enter_reason_list, exit_reason_list,
+                   indicator_list, columns=None):
+
+    if columns is None:
+        columns = ['pair', 'open_date', 'close_date', 'profit_abs', 'enter_reason', 'exit_reason']
+
+    bigdf = pd.DataFrame()
+    for pair, trades in analysed_trades[stratname].items():
+        bigdf = pd.concat([bigdf, trades], ignore_index=True)
+
+    if bigdf.shape[0] > 0 and ('enter_reason' in bigdf.columns):
+        if group is not None:
+            glist = group.split(",")
+            _do_group_table_output(bigdf, glist)
+
+        if enter_reason_list is not None and not enter_reason_list == "all":
+            enter_reason_list = enter_reason_list.split(",")
+            bigdf = bigdf.loc[(bigdf['enter_reason'].isin(enter_reason_list))]
+
+        if exit_reason_list is not None and not exit_reason_list == "all":
+            exit_reason_list = exit_reason_list.split(",")
+            bigdf = bigdf.loc[(bigdf['exit_reason'].isin(exit_reason_list))]
+
+        if indicator_list is not None:
+            if indicator_list == "all":
+                print(bigdf)
+            else:
+                available_inds = []
+                for ind in indicator_list.split(","):
+                    if ind in bigdf:
+                        available_inds.append(ind)
+                ilist = ["pair", "enter_reason", "exit_reason"] + available_inds
+                _print_table(bigdf[ilist], sortcols=['exit_reason'], show_index=False)
+        else:
+            _print_table(bigdf[columns], sortcols=['pair'], show_index=False)
+    else:
+        print("\\_ No trades to show")
```

Comment:
I seem to run into this consistently (i did run backtesting with `--export signals`).

the output is not helpful in determining what is wrong (i did use buy and sell reasons).

```
2022-05-26 19:28:59,221 - freqtrade.data.btanalysis - INFO - Loading backtest result from ../xmatt/backtest_results/backtest-result-2022-05-26_19-26-07.json
2022-05-26 19:28:59,228 - freqtrade.data.entryexitanalysis - INFO - Loaded signal candles: ../xmatt/backtest_results/backtest-result-2022-05-26_19-26-07_signals.pkl
2022-05-26 19:28:59,228 - freqtrade.data.entryexitanalysis - INFO - Processing None : 2 pairs
\_ No trades to show

```

Reply:
What's the backtest output?

Reply:
all i did was run `freqtrade analysis` without `--strategy` ...  (see other comments)

the backtest output itself was fine - but this ended up with backtestresult being empty (load error due to None as strategy) ...

Reply:
Using load_backtest_stats now